File: app/src/main/java/ru/ip_fateev/lavka/Inventory/ProductDao.java
package ru.ip_fateev.lavka.Inventory;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@Dao
public interface ProductDao {
    @Query("SELECT * FROM product")
    //LiveData<List<Product>> getAll();
    List<Product> getAll();

    @Query("SELECT * FROM product")
    LiveData<List<Product>> getAllLive();

    @Query("SELECT * FROM product WHERE id = :id")
    Product getById(long id);

    @Insert
    void insert(Product product);

    @Update
    void update(Product product);

    @Delete
    void delete(Product product);
}

File: app/src/main/java/ru/ip_fateev/lavka/data/ProductDao.kt
package ru.ip_fateev.lavka.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import kotlinx.coroutines.flow.Flow

@Dao
interface ProductDao {
    @Insert
    suspend fun insert(product: Product)

    @Update
    suspend fun update(product: Product)

    @Query("SELECT * FROM product ORDER BY id ASC")
    fun getFlow(): Flow<List<Product>>

    @Query("SELECT * FROM product ORDER BY id ASC")
    suspend fun get(): List<Product>

    @Query("SELECT * FROM product WHERE id = :id")
    fun getFlow(id: Long): Flow<List<Product>>

    @Query("SELECT * FROM product WHERE id = :id")
    suspend fun get(id: Long): List<Product>
}